src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -390,26 +407,19 @@ module.exports = function<T, P, I, TI, C>(
         }
         if (finishedWork.effectTag & Callback) {
           if (finishedWork.callbackList) {
-            const callbackError = callCallbacks(finishedWork.callbackList, instance);
-            firstError = firstError || callbackError;
+            const { callbackList } = finishedWork;
             finishedWork.callbackList = null;
+            callCallbacks(callbackList, instance);
```

Comment:
That's right. Didn't seem like it was worth the extra try-catch block, since the component will get unmounted anyway. I updated the test to match the new behavior.